fursquare/venues/views.py

```diff
@@ -0,0 +1,314 @@
+from django.shortcuts import render, get_object_or_404
+from django.http import HttpResponse, JsonResponse
+from django.views.decorators.csrf import csrf_exempt
+from django.contrib.auth.models import User
+from rest_framework.renderers import JSONRenderer
+from .models import Venue, Comment, Rating, VenueType
+from .serializers import VenueSerializer, CommentSerializer, VenueTypeSerializer, UserSerializer, RatingSerializer
+from rest_framework import status
+from rest_framework.decorators import api_view, permission_classes
+from rest_framework.response import Response
+from rest_framework.permissions import IsAuthenticated, IsAuthenticatedOrReadOnly, IsAdminUser
+from rest_framework.authtoken.models import Token
+from .permissions import IsAdminOrReadOnly, IsOwnerOrReadOnly
+from django.contrib.auth import authenticate
+
+
+@api_view(["POST"])
+def login(request):
+    username = request.data.get("username")
+    password = request.data.get("password")
+
+    user = authenticate(username=username, password=password)
+    if not user:
+        return Response({"error": "Login failed"}, status=status.HTTP_401_UNAUTHORIZED)
+    token = Token.objects.get_or_create(user=user)
+    return Response({"token": token.key})
+
+
+@api_view(['GET', 'POST'])
+@permission_classes((IsAdminOrReadOnly, ))
+def venue_list(request):
+    if request.method == 'GET':
+        venue_list = Venue.objects.all()
+        serializer = VenueSerializer(venue_list, many=True)
+        return Response(serializer.data)
+
+    elif request.method == 'POST':
+        data = request.data
+        user = request.user
+        data['created_by'] = user.id
+        serializer = VenueSerializer(data=data)
+        if serializer.is_valid():

```

Comment:
raise_exception=True'nun hepsi icin degismesi gerekiyor, hepsine tek tek yazmadim.

Reply:
hepsi için değiştirdiğimi düşünüyodum bu gözümden kaçmış